### PR TITLE
Added gui and test case to gameoflife

### PR DESCRIPTION
**Pull request by rtaub5**

### CODE REVIEW

File: src/main/java/taub/gameoflife/GameOfLifeGrid.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                g2.setColor(lightMagenta);
+                if (game.getOrigGameBoard()[row / unitMeasure][column] == 1)
+                {
+                    g2.setColor(darkMagenta);
+                }
+                g2.drawRect(startX, startY, unitMeasure, unitMeasure);
+                g2.fillRect(startX, startY, unitMeasure, unitMeasure);
+                startX += 25;
+            }
+             startY = startY + unitMeasure;
```
Comment (andrewoid) [inline src/main/java/taub/gameoflife/GameOfLifeGrid.java:43]: +=

File: src/main/java/taub/gameoflife/GameOfLifeGrid.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            int startX = 65;
+            for (int column = 0; column < game.getColumns(); column++)
+            {
+                g2.setColor(lightMagenta);
+                if (game.getOrigGameBoard()[row / unitMeasure][column] == 1)
+                {
+                    g2.setColor(darkMagenta);
+                }
+                g2.drawRect(startX, startY, unitMeasure, unitMeasure);
+                g2.fillRect(startX, startY, unitMeasure, unitMeasure);
```
Comment (andrewoid) [inline src/main/java/taub/gameoflife/GameOfLifeGrid.java]: If you're using fillRect you don't need drawRect

File: src/main/java/taub/gameoflife/GameOfLifeGrid.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        int unitMeasure = 25;
+        int startY = 55;
+
+        for (int row = 0; row < game.getRows() * unitMeasure; row = row + unitMeasure)
+        {
+            int startX = 65;
+            for (int column = 0; column < game.getColumns(); column++)
+            {
+                g2.setColor(lightMagenta);
```
Comment (andrewoid) [inline src/main/java/taub/gameoflife/GameOfLifeGrid.java]: put this in an `else`

File: src/main/java/taub/gameoflife/GameOfLifeGrid.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Override
+    public void paintComponent(Graphics g)
+    {
+        Graphics2D g2 = (Graphics2D) g;
+        Color lightMagenta = new Color(255, 0, 255, 25);
+        Color darkMagenta = new Color(255, 0, 255, 200);
+
+        g2.setColor(lightMagenta);
```
Comment (andrewoid) [inline src/main/java/taub/gameoflife/GameOfLifeGrid.java]: not necessary here

File: src/main/java/taub/gameoflife/GameFrame.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+
+        // setup actions
+        playButton.addActionListener(new ActionListener()
+        {
+            @Override
+            public void actionPerformed(ActionEvent e)
+            {
+                grid.game.nextGeneration();
+                grid.repaint();
```
Comment (andrewoid) [inline src/main/java/taub/gameoflife/GameFrame.java]: This should go in a timer, we can discuss this tomorrow.

File: src/test/java/taub/gameoflife/GameOfLifeTest.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        grid.nextGeneration();
+
+        //then
+        int[][] expectedBoard = {
+                {0, 1, 0},
+                {0, 1, 0},
+                {0, 1, 0}
+        };
+
+        assertEquals(boardToString(expectedBoard), grid.toString());
```
Comment (andrewoid) [inline src/test/java/taub/gameoflife/GameOfLifeTest.java]: return the grid and then you can compare grids by using `assertArrayEquals`

**Review by andrewoid — COMMENTED** [review]
Include a readme and a screenshot

File: src/main/java/taub/gameoflife/GameFrame.java (hunk 5 above)
Comment (rtaub5) [reply src/main/java/taub/gameoflife/GameFrame.java]: So I don't have to implement this yet? I should wait for class?

Comment (andrewoid) [reply src/main/java/taub/gameoflife/GameFrame.java]: You can do it now if you can figure it out.

File: src/main/java/taub/gameoflife/GameOfLifeGrid.java (hunk 2 above)
Comment (rtaub5) [reply src/main/java/taub/gameoflife/GameOfLifeGrid.java]: If you get rid of drawrect then there's no borders.

File: src/main/java/taub/gameoflife/GameFrame.java (hunk 5 above)
Comment (rtaub5) [reply src/main/java/taub/gameoflife/GameFrame.java]: Ok. I will create a pr without the timer for now, but I will work on it afterwards and see if I can figure it out.

File: src/main/java/taub/gameoflife/GameOfLifeGrid.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
+package taub.gameoflife;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class GameOfLifeGrid extends JComponent
+{
+    GameOfLife game;
```
Comment (andrewoid) [inline src/main/java/taub/gameoflife/GameOfLifeGrid.java]: private

**Review by andrewoid — APPROVED** [review]

**Pull request merged**